Sources/Apollo/GraphQLGETTransformer.swift

```diff
@@ -35,26 +32,23 @@ struct GraphQLGETTransformer {
       return nil
     }
     
-    guard let query = self.body.jsonObject[self.queryKey] as? String else {
+    var queryItems: [URLQueryItem] = []
+    
+    do {
+      _ = try body.compactMap({ arg in
+        if let value = arg.value as? GraphQLMap {
+          let data = try JSONSerialization.data(withJSONObject: value.jsonValue)
+          if let string = String(data: data, encoding: .utf8) {
+            queryItems.append(URLQueryItem(name: arg.key, value: string))
+          }
+        } else if let string = arg.value as? String {
+          queryItems.append(URLQueryItem(name: arg.key, value: string))
+        }
+      })
```

Comment:
So one thing I would be very cautious about with this change is the fact that it makes the query strings non-deterministic, as shown by the fact that you had to add additional strings to the tests. 

Can you explain why this is necessary?

Reply:
Oh, the original idea is assuming `GraphQLGETTransformer` should not have logic about the usage, so it has less dependency between APQs or other request, variables or query, but just convert the GraphQLMap to queryString with the support of multiple deep levels dictionary. As APQs requires 2 dimensions dict body.

To solve the `non-deterministic` issue which coming from the natural of dictionary. What do you think if sorting the allKeys first before constructing the queryItems array?

Reply:
Yeah I tried that. It unfortunately didn't help much.

I wound up creating some stuff in #628 that uses sorted keys where possible and then does tests to make sure stuff actually gets shoved in there when sorted keys are not possible. Would that work for you?

Reply:
Cool! will try on that method.

Reply:
updated with sorting and added a few test cases to cover 2lv GraphQLMap

Reply:
what do u mean by hand `body` straight to the `dataSortedIfPossible`

like this?
```
        let data = try JSONSerialization.dataSortedIfPossible(withJSONObject: body.jsonValue)
        if let string = String(data: data, encoding: .utf8) {
        
        }
```
you will got a string of the follow and can't pass as query string
`
"{\"extensions\":{\"persistedQuery\":{\"sha256Hash\":\"f6e76545cd03aa21368d9969cb39447f6e836a16717823281803778e7805d671\",\"version\":1}},\"query\":\"query HeroName($episode: Episode) {\\n  hero(episode: $episode) {\\n    __typename\\n    name\\n  }\\n}\",\"variables\":{\"episode\":null}}"
`

There will be query parameters of 
`query=xxx`, `variables=xxx`, `extensions=xxx`
and u can't  serialise it if `xxx` is already a string

Reply:
AHA. That's what I was missing. Thank you!